### PR TITLE
Benchmark

### LINKED ISSUE #2347: Updates to benchmark application
@tensorflow/micro 

resolves [b/297590035](https://issuetracker.google.com/297590035)
Add CSV output format

### PR DESCRIPTION
**Pull request by ddavis-2015**
@tensorflow/micro

Formatted column output
Added CSV output

resolves [b/297590035](https://issuetracker.google.com/297590035)

bug=fixes #2347

### CODE REVIEW

File: tensorflow/lite/micro/tools/benchmarking/metrics.cc [hunk 1]
```diff
@@ -15,88 +15,299 @@ limitations under the License.
...
+LogArenaRecord GetLogArenaRecord(
+    const tflite::RecordingSingleArenaBufferAllocator* allocator,
+    int row_index) {
+  const size_t total_bytes = allocator->GetUsedBytes();
+  const size_t allocations[] = {total_bytes,
+                                allocator->GetNonPersistentUsedBytes(),
+                                allocator->GetPersistentUsedBytes()};
+  static_assert(std::extent<decltype(allocations)>::value == kArenaRows,
+                "kArenaRows mismatch");
+  const char* titles[] = {"Total", "Head", "Tail"};
```
Comment (rascani) [inline tensorflow/lite/micro/tools/benchmarking/metrics.cc:62]: Let's switch this to: `{"Total", "NonPersistent", "Persistent"}`.

File: tensorflow/lite/micro/tools/benchmarking/metrics.cc [hunk 2]
```diff
@@ -15,88 +15,299 @@ limitations under the License.
...
+constexpr int kAllocationTypes = 7;
+constexpr int kAllocationColumns = 6;
 
-  PrintFormattedData<kArenaRows, kArenaCols>(headers, data, kArenaRows, type,
-                                             "Arena");
+constexpr int kMaxBufSize = 100;
+
+LogArenaRecord GetLogArenaRecord(
+    const tflite::RecordingSingleArenaBufferAllocator* allocator,
+    int row_index) {
```
Comment (rascani) [inline tensorflow/lite/micro/tools/benchmarking/metrics.cc:55]: We should probably add a DCHECK that row_index < kArenaRows

File: tensorflow/lite/micro/tools/benchmarking/metrics.cc [hunk 3]
```diff
@@ -15,88 +15,299 @@ limitations under the License.
...
+
+  return record;
 }
 
-void LogAllocations(const tflite::RecordingMicroAllocator& allocator,
-                    const PrettyPrintType type) {
-  constexpr int kAllocationTypes = 7;
-  tflite::RecordedAllocationType types[kAllocationTypes] = {
+LogAllocationRecord GetLogAllocationRecord(
+    const tflite::RecordingMicroAllocator& allocator, int row_index) {
```
Comment (rascani) [inline tensorflow/lite/micro/tools/benchmarking/metrics.cc:75]: Similar comment here about DCHECK(row_index < kAllocationTypes)

**Review by rascani — COMMENTED** [review]
LGTM, just a few nits.

File: tensorflow/lite/micro/tools/benchmarking/generic_model_benchmark.cc [hunk 4]
```diff
@@ -154,11 +151,31 @@ int Benchmark(const char* model_file_name) {
...
-int main(int argc, char** argv) { return tflite::Benchmark(argv[1]); }
+void usage(const char* prog_name) {
+  MicroPrintf("usage: %s filename [--csv]", prog_name);
+}
+
+int main(int argc, char** argv) {
+  // Which format should be used to output debug information.
+  tflite::PrettyPrintType print_type = tflite::PrettyPrintType::kTable;
+
+  if (--argc > 1) {
```
[the remark this answers is not in the document]
Comment (ddavis-2015) [reply tensorflow/lite/micro/tools/benchmarking/generic_model_benchmark.cc]: done

File: tensorflow/lite/micro/tools/benchmarking/metrics.cc (hunk 2 above)
Comment (ddavis-2015) [reply tensorflow/lite/micro/tools/benchmarking/metrics.cc:56]: done

File: tensorflow/lite/micro/tools/benchmarking/metrics.cc (hunk 1 above)
Comment (ddavis-2015) [reply tensorflow/lite/micro/tools/benchmarking/metrics.cc]: done

File: tensorflow/lite/micro/tools/benchmarking/metrics.cc (hunk 3 above)
Comment (ddavis-2015) [reply tensorflow/lite/micro/tools/benchmarking/metrics.cc:78]: done

**Review by rascani — APPROVED** [review]

**Pull request merged**